staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -950,6 +1100,20 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 	storageMap2[crd.UID] = ret
 	r.customStorage.Store(storageMap2)
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionAPI) ||
+		!utilfeature.DefaultFeatureGate.Enabled(features.APIServerIdentity) {
+		return ret, nil
+	}
+
+	// spawn storage version update in background and use channels to make handlers wait
+	processedCh := make(chan struct{})
+	abortedCh := make(chan struct{})
+	r.storageVersionManager.EnqueueStorageVersionUpdate(crd, nil, processedCh, abortedCh)
+	ret.storageVersionUpdate = &storageVersionUpdate{
```

Comment:
here is a race. The `r.customStorage.Store(storageMap2)` must come laster *AFTER* `storageMap2` and `ret` are in its final state.

Reply:
nice catch! Fixed